### ISSUE TITLE
Can not validate PackageInfo.g

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/anupq#13]
The following happens during check for updates:
```
* ANUPQ - new version 3.1.2 discovered!!!
 ============================================
 Getting new archives from 
 https://github.com/gap-packages/anupq/releases/download/v3.1.2/anupq-3.1.2
[ ".tar.bz2", ".tar.gz" ]
 unpacking anupq-3.1.2.tar.gz
#E  component `ArchiveURLSubset' must be bound to a list of strings denoting r\
elative paths to readable files or directories
#E  component `HTMLStart' must be bound to a string denoting a relative path t\
o a readable file
 ERROR (ANUPQ): validation of the info file not successful! SKIPPING!!!
```
Could you please make version 3.1.3 and fix this? You have
```
  ArchiveURLSubset := ["doc", "htm"],
  HTMLStart := "htm/chapters.htm",
```
which should be changed to
```
  ArchiveURLSubset := ["doc"],
  HTMLStart := "doc/chap0.html",
```
because of the apparent transition to GAPDoc (great!).

Note that this was avoidable by validating the URL of PackageInfo.g file using the form at http://www.gap-system.org/Packages/Authors/authors.html which reports:
```
File or directory htm containing documentation is missing in your package archive!

HTML help book start file (.HTMLStart) htm/chapters.htm is missing in your package archive!
```

**Comment by fingolfin** [comment]
Thanks for the note, I'll work on it this weekend or next week.

And yeah, it technically would have been avoidable, but it's very easy to forget that validation step, or do it incorrectly... Moreover, I can only use that validator form *after* uploading my package; that's is very annoying, because strictly speaking, at this point any change would already require a new release!

That's not an excuse, of course, just an observation. Anyway, easy solution: I'll try to improve `release` from my ReleaseTools to catch these errors.

**Comment by fingolfin** [comment]
Meta observation: I also think that validator is too hidden: I have to scroll waaay down on http://www.gap-system.org/Packages/Authors/authors.html and I also always forget that it is on that page. can't we move it to its own page, e.g.
  http://www.gap-system.org/Packages/Validator.html
and then link to it from multiple place? E.g. I would find it helpful if there was a link to it on http://www.gap-system.org/Packages/packages.html, ideally with the text "validator" in the link text. Because that's always the first place I search.

Meta Meta observation: If we could move GapWWW to github, then I could submit the above as an issue there, and possible open a pull request for it... Perhaps we can look into that next week? It might also be a first step towards a new website architecture?

**Comment by alex-konovalov** [comment]
Thanks. BTW, as the very http://www.gap-system.org/Packages/Authors/authors.html pages says, `ValidatePackageInfo(filename)` works from within GAP as well: see http://www.gap-system.org/Manuals/doc/ref/chap76.html#X79767C2482FF6F55. That's what is called in the script which output was copied above.

Perhaps `release` from ReleaseTools should include a call to `ValidatePackageInfo` ?

**Comment by alex-konovalov** [comment]
I am all for moving GapWWW and Mixer repos on GitHub. They are now Mercurial repositories hosted in St Andrews. Let's get this done next week?

**Comment by fingolfin** [comment]
Sure, it shoudl do that, but that's not enough, as `ValidatePackageInfo` only checks a fraction of the things (in fact, it passes for the anupq PackageInfo.g file). Anyway, see fingolfin/ReleaseTools#4

**Comment by alex-konovalov** [comment]
Yes - I think the key difference is whether you call `ValidatePackageInfo` on a record or on a string with a package name. Also, the GAP version behind the validator may be older than the latest release, so some checks may be missing there. As a result, sometimes the package may pass the online validator but get some warnings or even errors later.

**Comment by fingolfin** [comment]
This should be fixed in 3.1.3
